app/jobs/rest_of_world_zones/create_job.rb

```diff
@@ -0,0 +1,85 @@
+module RestOfWorldZones
+  class CreateJob < ApplicationJob
+    def perform(shopify_shipping_zone)
+      @shopify_shipping_zone = shopify_shipping_zone
+      @existed_spree_zones_by_member_type = existed_spree_zones_by_member_type
+      @profile_id = profile_id
+      create_rest_of_world_country_zone
+    end
+
+    private
+
+    def existed_spree_zones_by_member_type
+      Spree::Zone.where("name like ?", "%#{@profile_id}%").map(&:members).flatten.group_by{ |m| m.zoneable_type }
+    end
+
+    def profile_id
+      @profile_id = @shopify_shipping_zone.profile_id.split('/').last
```

Comment:
you have two assigns to the same instance variable, here and on the line `6`
I suggest to assign it directly in perform method and remove this method

Reply:
yes, I did it, and later changed sth, sorry ;)